File: hello-world/hello.jl.lua.rb
#!/usr/bin/env ruby
_=#_G --[[
0 #= Code visible only to Ruby

puts "Hello from Ruby!"

=begin Code visible only to Julia =#

puts = println
puts("Hello from Julia!")

#= Code visible only to Lua ]]

puts = print
puts("Hello from Lua!")

--[[
=end # Code visible to Julia, Lua and Ruby =##]]
puts("Hello from \x1b[35mJulia, \x1b[94mLua\x1b[m and \x1b[91mRuby\x1b[m!")

File: hello-world/hello_jl_lua.py
#!/usr/bin/env lua
(#{} and ""):byte() --[[
""" Code visible only to Lua ]]

println = print
println("Hello from \x1b[94mLua\x1b[m!")

--[[ Code visible only to Python """)#=

println = print
println("Hello from \x1b[36mPython\x1b[m!")

""" Code visible only to Julia =#

println("Hello from \x1b[35mJulia\x1b[m!")

# Code visible to Julia, Lua and Python """#]]

println("Hello from \x1b[35mJulia\x1b[m, \x1b[94mLua\x1b[m and \x1b[36mPython\x1b[m!")
